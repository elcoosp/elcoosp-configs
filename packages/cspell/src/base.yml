dictionaryDefinitions:
  - name: ts-libs
    path: ./ts-libs.txt
    addWords: true
dictionaries:
  - ts-libs
ignorePaths:
  - node_modules
  - pnpm-lock.yaml
  - ./ts-libs.txt
