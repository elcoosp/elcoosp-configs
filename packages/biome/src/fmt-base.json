{
  "organizeImports": {
    "enabled": true
  },
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true
  },
  "formatter": {
    "indentStyle": "space",
    "ignore": ["__snapshots__"]
  },
  "javascript": {
    "formatter": {
      "quoteStyle": "single",
      "semicolons": "asNeeded"
    }
  },
  "css": {
    "parser": {
      "cssModules": true
    }
  },
  "linter": {
    "enabled": false
  }
}
