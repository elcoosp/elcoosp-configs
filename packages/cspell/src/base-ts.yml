dictionaryDefinitions:
  - name: ts-libs
    path: ./ts-libs.txt
    addWords: true
dictionaries:
  - ts-libs
  - typescript
  - node
  - npm
ignorePaths:
  - node_modules
  - pnpm-lock.yaml
  - ./ts-libs.txt
