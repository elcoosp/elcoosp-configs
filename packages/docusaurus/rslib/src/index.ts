import { type RslibConfig, defineConfig } from "@rslib/core";
export type Config = {
	preset: "dual";
};
const presets = {
	dual: {
		lib: [
			{
				format: "esm",
				syntax: "es2021",
				dts: true,
			},
			{
				format: "cjs",
				output: {
					distPath: {
						root: "./dist/cjs",
					},
				},
			},
		],
	},
} as const satisfies Record<Config["preset"], RslibConfig>;
export function createRsLibConfig({ preset }: Config) {
	return defineConfig(presets[preset]);
}
