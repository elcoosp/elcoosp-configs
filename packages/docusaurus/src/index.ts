import type * as Preset from '@docusaurus/preset-classic';
import type { Config } from '@docusaurus/types';
import { deepmerge } from 'deepmerge-ts';
import { themes as prismThemes } from 'prism-react-renderer';
import { titleCase } from 'title-case';

type DerivedConfig = {
  githubUrl: string;
} & Pick<Config, 'organizationName' | 'title'> &
  StaticConfig;
type StaticConfig = {
  projectName: string;
} & Pick<Config, 'organizationName'>;
export function themeConfig({
  title,
  githubUrl,
  projectName,
}: DerivedConfig): Preset.ThemeConfig {
  return {
    image: 'img/social-card.png',
    navbar: {
      title,
      logo: {
        alt: `${title} logo`,
        src: 'img/logo.svg',
      },
      items: [
        {
          type: 'docSidebar',
          sidebarId: 'docsSidebar',
          position: 'left',
          label: 'Docs',
        },
        { to: '/blog', label: 'Blog', position: 'left' },
        {
          href: githubUrl,
          label: 'GitHub',
          position: 'right',
        },
      ],
    },
    footer: {
      style: 'dark',
      links: [
        {
          title: 'Docs',
          items: [
            {
              label: 'Docs',
              to: '/docs/intro',
            },
          ],
        },
        {
          title: 'Community',
          items: [
            {
              label: 'Stack Overflow',
              href: `https://stackoverflow.com/questions/tagged/${projectName}`,
            },
          ],
        },
        {
          title: 'More',
          items: [
            {
              label: 'Blog',
              to: '/blog',
            },
            {
              label: 'GitHub',
              href: githubUrl,
            },
          ],
        },
      ],
    },
    prism: {
      theme: prismThemes.github,
      darkTheme: prismThemes.dracula,
    },
  };
}
export function classicPreset(
  config: DerivedConfig,
): ['classic', Preset.Options] {
  return [
    'classic',
    {
      docs: {
        sidebarPath: './sidebars.ts',
        editUrl: `${config.githubUrl}/tree/main/documentation/blog`,
      },
      blog: {
        showReadingTime: true,
        feedOptions: {
          type: ['rss', 'atom'],
          xslt: true,
        },
        editUrl: `${config.githubUrl}/tree/main/documentation/docs`,
        onInlineTags: 'warn',
        onInlineAuthors: 'warn',
        onUntruncatedBlogPosts: 'warn',
      },
      theme: {
        customCss: './src/css/custom.css',
      },
    } satisfies Preset.Options,
  ];
}

export function createDocusaurusConfig(
  { projectName, organizationName }: StaticConfig,
  getConfig: (derivedConfig: DerivedConfig) => Partial<Config>,
): Config {
  const githubUrl = `https://github.com/${organizationName}/${projectName}`;
  const deploy = 'vercel' as const;
  const url = { vercel: `https://${projectName}.vercel.app` }[deploy];
  const title = titleCase(projectName);
  const baseUrl = { vercel: '/' }[deploy];
  return deepmerge<[Config, Config]>(
    getConfig({
      githubUrl,
      organizationName,
      title,
      projectName,
    }) as Config,
    {
      url,
      baseUrl,
      organizationName,
      title,
      favicon: 'img/favicon.ico',
      onBrokenLinks: 'throw',
      onBrokenMarkdownLinks: 'warn',
      i18n: {
        defaultLocale: 'en',
        locales: ['en'],
      },
      future: {
        experimental_faster: true,
      },
    },
  );
}
