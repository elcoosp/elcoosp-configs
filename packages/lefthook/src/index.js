console.error(
  'There is no js files exported in this package, check the src/*.yml lefthook preset files',
);
