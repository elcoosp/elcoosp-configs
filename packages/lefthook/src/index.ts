export const ERROR_MESSAGE =
  'There is no js files exported in this package, check the src/*.yml lefthook preset files';
console.error(ERROR_MESSAGE);
