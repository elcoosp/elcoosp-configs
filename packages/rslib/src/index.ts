import type { RslibConfig } from '@rslib/core';
import { readPackage } from 'read-pkg';
export type Config = {
  preset: 'dual';
};
const presets = {
  dual: async () => {
    const pkg = await readPackage();
    // TODO check that match inside package.json
    const cjsOutDistPathRoot = './dist/cjs';
    return {
      lib: [
        {
          format: 'esm',
          syntax: 'es2021',
          dts: true,
        },
        {
          format: 'cjs',
          output: {
            distPath: {
              root: cjsOutDistPathRoot,
            },
          },
        },
      ],
    };
  },
} as const satisfies Record<Config['preset'], () => Promise<RslibConfig>>;
export async function createRsLibConfig({
  preset,
}: Config): Promise<RslibConfig> {
  // Should not call defineConfig otherwise build fail
  return await presets[preset]();
}
