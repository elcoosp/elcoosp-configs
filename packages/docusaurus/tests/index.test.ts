import { expect, test } from "vitest";
import { classicPreset, createDocusaurusConfig, themeConfig } from "../src";

test("createDocusaurusConfig", () => {
	const result = createDocusaurusConfig(
		{ organizationName: "elcoosp", projectName: "readma" },
		(derived) => {
			return {
				tagline: "Beautiful effortless README generator",
				plugins: [],
				presets: [classicPreset(derived)],
				themeConfig: themeConfig(derived),
			};
		},
	);
	expect(result).toMatchSnapshot();
});
