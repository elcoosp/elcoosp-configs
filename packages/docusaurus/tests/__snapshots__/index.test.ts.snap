// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createDocusaurusConfig 1`] = `
{
  "baseUrl": "/",
  "favicon": "img/favicon.ico",
  "future": {
    "experimental_faster": true,
  },
  "i18n": {
    "defaultLocale": "en",
    "locales": [
      "en",
    ],
  },
  "onBrokenLinks": "throw",
  "onBrokenMarkdownLinks": "warn",
  "organizationName": "elcoosp",
  "plugins": [],
  "presets": [
    [
      "classic",
      {
        "blog": {
          "editUrl": "https://github.com/elcoosp/readma/tree/main/documentation/docs",
          "feedOptions": {
            "type": [
              "rss",
              "atom",
            ],
            "xslt": true,
          },
          "onInlineAuthors": "warn",
          "onInlineTags": "warn",
          "onUntruncatedBlogPosts": "warn",
          "showReadingTime": true,
        },
        "docs": {
          "editUrl": "https://github.com/elcoosp/readma/tree/main/documentation/blog",
          "sidebarPath": "./sidebars.ts",
        },
        "theme": {
          "customCss": "./src/css/custom.css",
        },
      },
    ],
  ],
  "tagline": "Beautiful effortless README generator",
  "themeConfig": {
    "footer": {
      "links": [
        {
          "items": [
            {
              "label": "Docs",
              "to": "/docs/intro",
            },
          ],
          "title": "Docs",
        },
        {
          "items": [
            {
              "href": "https://stackoverflow.com/questions/tagged/readma",
              "label": "Stack Overflow",
            },
          ],
          "title": "Community",
        },
        {
          "items": [
            {
              "label": "Blog",
              "to": "/blog",
            },
            {
              "href": "https://github.com/elcoosp/readma",
              "label": "GitHub",
            },
          ],
          "title": "More",
        },
      ],
      "style": "dark",
    },
    "image": "img/social-card.png",
    "navbar": {
      "items": [
        {
          "label": "Docs",
          "position": "left",
          "sidebarId": "docsSidebar",
          "type": "docSidebar",
        },
        {
          "label": "Blog",
          "position": "left",
          "to": "/blog",
        },
        {
          "href": "https://github.com/elcoosp/readma",
          "label": "GitHub",
          "position": "right",
        },
      ],
      "logo": {
        "alt": "Readma logo",
        "src": "img/logo.svg",
      },
      "title": "Readma",
    },
    "prism": {
      "darkTheme": {
        "plain": {
          "backgroundColor": "#282A36",
          "color": "#F8F8F2",
        },
        "styles": [
          {
            "style": {
              "color": "rgb(189, 147, 249)",
            },
            "types": [
              "prolog",
              "constant",
              "builtin",
            ],
          },
          {
            "style": {
              "color": "rgb(80, 250, 123)",
            },
            "types": [
              "inserted",
              "function",
            ],
          },
          {
            "style": {
              "color": "rgb(255, 85, 85)",
            },
            "types": [
              "deleted",
            ],
          },
          {
            "style": {
              "color": "rgb(255, 184, 108)",
            },
            "types": [
              "changed",
            ],
          },
          {
            "style": {
              "color": "rgb(248, 248, 242)",
            },
            "types": [
              "punctuation",
              "symbol",
            ],
          },
          {
            "style": {
              "color": "rgb(255, 121, 198)",
            },
            "types": [
              "string",
              "char",
              "tag",
              "selector",
            ],
          },
          {
            "style": {
              "color": "rgb(189, 147, 249)",
              "fontStyle": "italic",
            },
            "types": [
              "keyword",
              "variable",
            ],
          },
          {
            "style": {
              "color": "rgb(98, 114, 164)",
            },
            "types": [
              "comment",
            ],
          },
          {
            "style": {
              "color": "rgb(241, 250, 140)",
            },
            "types": [
              "attr-name",
            ],
          },
        ],
      },
      "theme": {
        "plain": {
          "backgroundColor": "#f6f8fa",
          "color": "#393A34",
        },
        "styles": [
          {
            "style": {
              "color": "#999988",
              "fontStyle": "italic",
            },
            "types": [
              "comment",
              "prolog",
              "doctype",
              "cdata",
            ],
          },
          {
            "style": {
              "opacity": 0.7,
            },
            "types": [
              "namespace",
            ],
          },
          {
            "style": {
              "color": "#e3116c",
            },
            "types": [
              "string",
              "attr-value",
            ],
          },
          {
            "style": {
              "color": "#393A34",
            },
            "types": [
              "punctuation",
              "operator",
            ],
          },
          {
            "style": {
              "color": "#36acaa",
            },
            "types": [
              "entity",
              "url",
              "symbol",
              "number",
              "boolean",
              "variable",
              "constant",
              "property",
              "regex",
              "inserted",
            ],
          },
          {
            "style": {
              "color": "#00a4db",
            },
            "types": [
              "atrule",
              "keyword",
              "attr-name",
              "selector",
            ],
          },
          {
            "style": {
              "color": "#d73a49",
            },
            "types": [
              "function",
              "deleted",
              "tag",
            ],
          },
          {
            "style": {
              "color": "#6f42c1",
            },
            "types": [
              "function-variable",
            ],
          },
          {
            "style": {
              "color": "#00009f",
            },
            "types": [
              "tag",
              "selector",
              "keyword",
            ],
          },
        ],
      },
    },
  },
  "title": "Readma",
  "url": "https://readma.vercel.app",
}
`;
